sdk/eventgrid/eventgrid/review/eventgrid.api.md

```diff
@@ -0,0 +1,206 @@
+import { KeyCredential } from '@azure/core-auth';
+import { OperationOptions } from '@azure/core-http';
+import { PipelineOptions } from '@azure/core-http';
+import { RestResponse } from '@azure/core-http';
+
+export { AzureKeyCredential }
+
+// @public
+export interface CloudEvent<T> {
+    [keyof: string]: any;
+    data?: T;
+    datacontenttype?: string;
+    dataschema?: string;
+    id: string;
+    source: string;
+    subject?: string;
+    time?: Date;
+    type: string;
+}
+
+// @public
+export interface ContainerRegistryArtifactEventData {
+    action?: string;
+    id?: string;
+    target?: ContainerRegistryArtifactEventTarget;
+    timestamp?: Date;
+}
+
+// @public
+export interface ContainerRegistryArtifactEventTarget {
+    digest?: string;
+    mediaType?: string;
+    name?: string;
+    repository?: string;
+    size?: number;
+    tag?: string;
+    version?: string;
+}
+
+// @public
+export type ContainerRegistryChartDeletedEventData = ContainerRegistryArtifactEventData & {};
+
+// @public
+export type ContainerRegistryChartPushedEventData = ContainerRegistryArtifactEventData & {};
+
+// @public
+export interface ContainerRegistryEventActor {
+    name?: string;
+}
+
+// @public
+export interface ContainerRegistryEventData {
+    action?: string;
+    actor?: ContainerRegistryEventActor;
+    id?: string;
+    request?: ContainerRegistryEventRequest;
+    source?: ContainerRegistryEventSource;
+    target?: ContainerRegistryEventTarget;
+    timestamp?: Date;
+}
+
+// @public
+export interface ContainerRegistryEventRequest {
+    addr?: string;
+    host?: string;
+    id?: string;
+    method?: string;
+    useragent?: string;
+}
+
+// @public
+export interface ContainerRegistryEventSource {
+    addr?: string;
+    instanceID?: string;
+}
+
+// @public
+export interface ContainerRegistryEventTarget {
+    digest?: string;
+    length?: number;
+    mediaType?: string;
+    repository?: string;
+    size?: number;
+    tag?: string;
+    url?: string;
+}
+
+// @public
+export type ContainerRegistryImageDeletedEventData = ContainerRegistryEventData & {};
+
+// @public
+export type ContainerRegistryImagePushedEventData = ContainerRegistryEventData & {};
+
+// @public
+export class EventGridClient {
+    constructor(endpointUrl: string, credential: KeyCredential | SignatureCredential, options?: EventGridClientOptions);
+    readonly apiVersion: string;
+    readonly endpointUrl: string;
+    generateSharedAccessSignature(expiresOnUtc: Date): string;
+    sendCloudEvents(events: CloudEvent<any>[], options?: SendCloudEventsOptions): Promise<RestResponse>;
+    sendCustomSchemaEvents(events: Record<string, any>[], options?: SendCustomSchemaEventsOptions): Promise<RestResponse>;
+    sendEvents(events: EventGridEvent<any>[], options?: SendEventsOptions): Promise<RestResponse>;
+}
+
+// @public
+export type EventGridClientOptions = PipelineOptions;
+
+// @public
+export class EventGridConsumer {
```

Comment:
Is this talking to the service? If so, per guidelines this should likely be EventGridConsumerClient I think? There are parallels here with SB/EH, so pinging @richardpark-msft for this thoughts 😀

Reply:
> Is this talking to the service? 

It is not, which is why I explicitly made the choice to not include the `Client` suffix.  The Event Grid service has no API for "receiving" events.  Instead the service delivers messages to consumers as at they are published (and these consumers can be many different kinds of services, like invoking a webhook with the event data, putting the event data in a storage queue, or putting the event data in a message in a service bus queue or topic).  The type here is used to help you decode that data once you've received it from another one of these services.